### PR TITLE
[MAMAEdu-122693] Enroll to `invitation_only` course through API

### PR DESCRIPTION
**Pull request by dyudyunov**
(with email)

**Youtrack:**
https://youtrack.raccoongang.com/issue/MAMAEdu-122693

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: common/djangoapps/enrollment/tests/test_views.py [hunk 1]
```diff
@@ -210,22 +211,30 @@ def test_enroll(self, course_modes, enrollment_mode):
         self.assertTrue(is_active)
         self.assertEqual(course_mode, enrollment_mode)
 
-    def test_enroll_by_email(self):
+    def test_enroll_by_email(self, invitation_only=False):
         """
         Test enroll through API by email.
         """
+        if invitation_only is True:
```
Comment (OPersian) [inline common/djangoapps/enrollment/tests/test_views.py:218]: `if invitation_only:` might suffice I guess

File: common/djangoapps/enrollment/tests/test_views.py [hunk 2]
```diff
@@ -210,22 +211,30 @@ def test_enroll(self, course_modes, enrollment_mode):
...
+            # use course with `invitation_only=True`
+            self.course.invitation_only = True
         CourseModeFactory.create(
             course_id=self.course.id,
             mode_slug=CourseMode.VERIFIED,
             mode_display_name=CourseMode.VERIFIED,
         )
-
+        if invitation_only is True:
```
Comment (OPersian) [inline common/djangoapps/enrollment/tests/test_views.py:226]: same here:
`if invitation_only:`

File: common/djangoapps/enrollment/tests/test_views.py (hunk 1 above)
Comment (GlugovGrGlib) [reply common/djangoapps/enrollment/tests/test_views.py:218]: In my opinion it's better to explicitly check boolean, particularly in this function invitation_only may be any type

File: common/djangoapps/enrollment/tests/test_views.py [hunk 3]
```diff
@@ -210,22 +211,30 @@ def test_enroll(self, course_modes, enrollment_mode):
...
         self.assertEqual(course_mode, enrollment_mode)
 
-    def test_enroll_by_email(self):
+    def test_enroll_by_email(self, invitation_only=False):
         """
         Test enroll through API by email.
         """
+        if invitation_only is True:
+            # use course with `invitation_only=True`
+            self.course.invitation_only = True
```
Comment (GlugovGrGlib) [inline common/djangoapps/enrollment/tests/test_views.py:220]: Instead of if, there can be self.course.invitation_only = bool(invitation_only)

**Pull request merged**